boptimol/optimizer.py

```diff
@@ -0,0 +1,114 @@
+"""Methods for optimizing using internal coordinates"""
+import logging
+from csv import DictWriter
+from datetime import datetime
+from io import StringIO
+from pathlib import Path
+from typing import List, Optional
+
+import chemcoord as cc
+import torch
+import numpy as np
+
+from boptimol import Molecule
+from boptimol.GP_MODEL import select_next_points_botorch
+from boptimol.initializer import initial_guess
+
+logger = logging.getLogger(__name__)
+
+
+def run_optimization(mol: Molecule, n_steps: int, 
+                     init_steps: int, out_dir: Optional[Path]) -> np.ndarray:
+    
+    """Optimize the structure of a molecule by iteratively changing the dihedral angles
```

Comment:
Iteratively changing the z-matrix parameters. (I think I copied this from the conformer optimizer which only optimizes dihedrals.)